sunpy/map/sources/sdo.py

```diff
@@ -107,24 +103,22 @@ def __init__(self, data, header, **kwargs):
 
         GenericMap.__init__(self, data, header, **kwargs)
 
-        self.meta['detector'] = "HMI"
-#        self.meta['instrme'] = "HMI"
-#        self.meta['obsrvtry'] = "SDO"
+        self.meta['detector'] = self.meta.get('detector') or "HMI"
```

Comment:
Would `self.meta.get('detector', "HMI")` not work?

Reply:
Yes that would be much better.